src/envoy/mixer/mixer_control.cc

```diff
@@ -382,6 +388,32 @@ void MixerControl::BuildTcpReport(
       Attributes::TimeValue(std::chrono::system_clock::now());
 }
 
+// Mesh attributes from Pilot are all string type.
+// The attributes with ".ip" suffix will be treated
+// as ipv4 and use BYTES attribute type.
+void MixerControl::SetMeshAttribute(const std::string& name,
+                                    const std::string& value,
+                                    Attributes* attr) const {
+  // Check with ".ip" suffix,
+  if (name.length() < kIPSuffix.length() ||
```

Comment:
"name" can be ".ip"? otherwise '<=' seems to be better,
Does name.compare need to be case insensitive?

Reply:
Thanks.  ".ip" is not allowed,  code is changed.
All istio attribute names are lower cases.  
https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html